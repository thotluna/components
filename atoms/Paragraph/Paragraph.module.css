.paragraph {
  width: 100%;
  text-align: justify;
}

.size-sm {
  font-size: var(--font-size-sm);
}

.size-md {
  font-size: var(--font-size-md);
}

.size-lg {
  font-size: var(--font-size-lg);
}

.color-base {
  color: var(--color-font-base);
}

.color-primary {
  color: var(--color-font-primary);
}

.color-inverted {
  color: var(--color-base-white);
}

.is-inline {
  width: auto;
}
