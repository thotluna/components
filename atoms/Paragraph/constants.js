export const options = {
  colors: ['base', 'primary', 'inverted'],
  sizes: ['sm', 'md', 'lg'],
}
