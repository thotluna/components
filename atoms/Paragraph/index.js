export { default } from './Paragraph'
export { options } from './constants'
export { default as styles } from './Paragraph.module.css'
