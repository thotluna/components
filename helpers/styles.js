import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
