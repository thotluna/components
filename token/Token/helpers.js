import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
