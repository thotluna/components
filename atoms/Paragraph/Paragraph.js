import React from 'react'
import PropTypes from 'prop-types'
import { styles } from './Paragraph.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'

export const Paragraph = ({ children, isInline, getStyles }) => {
  return (
    <p
      className={getStyles('container', ['color', 'size'], {
        'is-inline': isInline,
      })}
    >
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  children: PropTypes.node,
  getStyles: PropTypes.func.isRequired,
  isInline: PropTypes.bool,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.size),
}

Paragraph.defaultProps = {
  isInline: false,
  color: 'base',
  size: 'md',
}

export default withStyles(styles)(Paragraph)
