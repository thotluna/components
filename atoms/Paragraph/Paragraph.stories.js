import React from 'react'
import Paragraph, { options } from '.'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <Paragraph {...args} />
const ListTemplate = getVariusTemplate(Paragraph)

export default {
  title: 'Atoms/Paragraph',
  component: Paragraph,
  args: {
    children:
      "I can't remember anything Can't tell if this is true or dream deep down inside I feel the scream This terrible silence stops in me Now that the war is through with me I'm waking up, I cannot see That there is not much left of me Nothing is real but pain now",
  },
  argTypes: {
    children: { control: 'text' },
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.size),
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Size = ListTemplate.bind({})
Size.args = { items: options.size.map((size) => ({ size })) }

export const IsInline = Template.bind({})
IsInline.args = { isInline: true, children: 'it is paragraph inline' }
