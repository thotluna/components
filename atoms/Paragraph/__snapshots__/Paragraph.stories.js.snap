// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Paragraph Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <p
    className="paragraph color-base size-sm"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
  <p
    className="paragraph color-primary size-sm"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
  <p
    className="paragraph color-inverted size-sm"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
</div>
`;

exports[`Storyshots Atoms/Paragraph Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <p
    className="paragraph color-base size-sm"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
</div>
`;

exports[`Storyshots Atoms/Paragraph Size 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <p
    className="paragraph color-base size-sm"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
  <p
    className="paragraph color-base size-md"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
  <p
    className="paragraph color-base size-lg"
  >
    Consent doubt Tuckborough challenge destroying.
  </p>
</div>
`;
